app/questionnaire/questionnaire_schema.py

```diff
@@ -961,6 +961,20 @@ def get_item_anchor(self, section_id: str, list_name: str) -> Optional[str]:
                 if item["for_list"] == list_name and item.get("item_anchor_answer_id"):
                     return f"#{str(item['item_anchor_answer_id'])}"
 
+    def get_placeholder_dependencies(self, transform_name: str) -> list[str | None]:
+        section_ids = []
+        mapping = get_mappings_with_key("transform", self.json.get("sections"))
+        for transform in mapping:
+            if transform["transform"] == transform_name:
+                for item in transform.get("arguments", []).get("items", []):
+                    if item.get("source") == "answers":
+                        block = self.get_block_for_answer_id(item["identifier"])
+                        section_id = self.get_section_id_for_block_id(block["id"])  # type: ignore
+                        if section_id not in section_ids:
+                            section_ids.append(section_id)
+
+        return section_ids
```

Comment:
Couple of things:
1. You need to evaluate the dependencies up front and not at run time. This can be pre-determined at load time. It is wasteful to fetch this every time, for a given schema the dependency won't change.
2. This is fetching the dependency for the entire survey, it should be by "block_id" of the current location and if you haven't got a block_id (i.e summary), then by section. Have a look at how the calculated summary one is done in placeholder, it needs to be similar to that. At the moment, if section 2 depends on section 1, and section 4 depends on section 3. Section 2 needs section 1's path, section 4 needs section 3's path, but the current implementation would mean section 2 and 4 will have section 1 and section 3's path which is wasteful and inefficient.

Reply:
Updated to evaluate dependencies on load time 👍

Reply:
Using block_id of current location now 👍

Reply:
Resolving on behalf of Mebin